source/common/upstream/health_checker_base_impl.cc

```diff
@@ -353,5 +378,38 @@ void HealthCheckEventLoggerImpl::logAddHealthy(
   file_->write(fmt::format("{}\n", json));
 }
 
+void HealthCheckEventLoggerImpl::logDegraded(
+    envoy::data::core::v2alpha::HealthCheckerType health_checker_type,
+    const HostDescriptionConstSharedPtr& host) {
+  envoy::data::core::v2alpha::HealthCheckEvent event;
```

Comment:
could we take all of the logging functions common attribute sets and move them to a common function? Similar to what @nezdolik is doing in #5517 with `setCommonEventParams`

Reply:
I went a bit further and also DRYd up the json serialization and writing, let me know what you think